### PR TITLE
Display Razorpay icon in saved payment methods

### PR DESCRIPTION
**Pull request by billrobbins**
Closes: https://github.com/Automattic/payments-florin/issues/435

## Proposed Changes

In adding support for Razorpay, we need to improve how saved payment methods are displayed to customers.  Currently, a placeholder credit card icon and `razorpay` are what visitors see.  This PR changes that by replacing the icon with the Razorpay logo and having `Razorpay` displayed if an email address is not available.

**Before**
razorpay-before

**After**
razorpay-payment-method

## Testing Instructions

Taken from https://github.com/Automattic/wp-calypso/pull/86197

* (a12s) Enable store sandbox and billing sandbox.
* Change your user's currency to INR.
* In checkout, set your tax location to India. Use any 6-digit numeric postal code.
* After clicking to submit in checkout, a modal window will appear.
 * If prompted to enter a phone number, use 7513200000.
 * You will be presented with a list of payment methods. Docs about testing these is [here](https://razorpay.com/docs/payments/payment-gateway/ecommerce-plugins/wordpress/test-integration/).
 * Card: Use 4111 1111 1111 1111 with any future expiration date, any 3 digit cvv, and any 4 digit 2FA code to simulate a successful payment.
* Once payment has been successful, go to `me/purchases/payment-methods` and you should see the Razorpay logog next to the payment method.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/assets/images/upgrades/razorpay.svg
Comment (billrobbins) [inline client/assets/images/upgrades/razorpay.svg:1]: The SVG is available at https://github.com/Automattic/wp-calypso/blob/52a957fb72f1356e9eb8ccf84a61c9fd83601b00/packages/wpcom-checkout/src/payment-methods/razorpay.tsx#L98 however it is a JSX element and `payment-methods.tsx` needs a string.

File: client/lib/checkout/payment-methods.tsx
Comment (ignaciodob) [inline client/lib/checkout/payment-methods.tsx]: Nitpick, from what I'm reading on the docs it's just "emandates" or "mandates"

File: client/lib/checkout/payment-methods.tsx
Comment (ignaciodob) [inline client/lib/checkout/payment-methods.tsx]: Maybe an email is not appropriate here since the accounts in Razorpay will be tied to the WPCOM users' email accounts either way (whereas in Paypal, the email account can be different)

**Review by ignaciodob — APPROVED** [review]
This looks good to me! Works well too! I left two non-blockers comments

Once we merge the changes for recurring payments, we can add a bit more information to this element, like the selected Payment Method (UPI) or the e-mandate token id.

File: client/lib/checkout/payment-methods.tsx
Comment (billrobbins) [reply client/lib/checkout/payment-methods.tsx]: I had wondered about that.  Is there anything you'd like to see here other than Razorpay?

Comment (ignaciodob) [reply client/lib/checkout/payment-methods.tsx]: Great question, but I don't think we have anything relevant to add here right now. Once we merge the changes for recurring payments, we can add the payment method used, e.g., UPI, credit card, e-mandate, etc.

**Comment by nbloomf** [comment]
The code looks great. My only question is whether customers will know what Razorpay means here -- for instance, with PayPal folks tend to think in terms of "paying with PayPal", but Razorpay is a vendor that implements the underlying UPI protocol for emandates, so I'm not sure if people think in terms of Razorpay, UPI, or emandates (or something else). We'll need to get input from someone in India; fortunately we know a guy. ;)

@kanishkdudeja -- when a user pays for a subscription using UPI, we show them the mandate in their list of stored payment details since it can be used for renewals. (I think non-UPI details will not be shown here because they are not rechargeable, like e.g. sofort and giropay). What name for this entity would be the most understandable for customers? Is the Razorpay brand like PayPal, where end users interact with it independently and think of it as its own payment method? Or is it more like Stripe, where the user has a relationship with the bank that issued their card, but not so much with Stripe, such that if we showed the Stripe logo here it would be confusing?

**Comment by kanishkdudeja** [comment]
> @kanishkdudeja -- when a user pays for a subscription using UPI, we show them the mandate in their list of stored payment details since it can be used for renewals. (I think non-UPI details will not be shown here because they are not rechargeable, like e.g. sofort and giropay). What name for this entity would be the most understandable for customers? Is the Razorpay brand like PayPal, where end users interact with it independently and think of it as its own payment method? Or is it more like Stripe, where the user has a relationship with the bank that issued their card, but not so much with Stripe, such that if we showed the Stripe logo here it would be confusing?

Good question @nbloomf! While the Razorpay brand is a fair bit popular in India, I think users in India tend to think of subscriptions linked to either their cards (credit/debit) or their UPI handles.

So from that perspective, I would recommend showing the [UPI logo](https://upload.wikimedia.org/wikipedia/commons/thumb/e/e1/UPI-Logo-vector.svg/400px-UPI-Logo-vector.svg.png) for the saved payment methods.

**Comment by billrobbins** [comment]
@kanishkdudeja is this more of what you have in mind here?
UPI

**Comment by kanishkdudeja** [comment]
> @kanishkdudeja is this more of what you have in mind here?

Yes, that looks about right to me :)

File: client/lib/checkout/payment-methods.tsx
Comment (billrobbins) [inline client/lib/checkout/payment-methods.tsx:11]: I'm assigning the UPI logo to `razorpayImage` here since we're using it for the logo instead of the actual Razorpay logo.

File: client/lib/checkout/payment-methods.tsx
Comment (michaeldcain) [inline client/lib/checkout/payment-methods.tsx]: @kanishkdudeja @billrobbins should this be `Unified Payments Interface (UPI)` or maybe `Razorpay UPI`?  I don't know which would be the more common way to refer to it in India, but although Stripe is pretty common, we tend to leave the processor out of this label. The icon helps, but we should assume that it might not be shown.

Comment (billrobbins) [reply client/lib/checkout/payment-methods.tsx]: I'll defer to @kanishkdudeja since he has the best context on this.

**Review by sirbrillig — APPROVED** [review]
Apart from Cain's comment about translations which I leave to you to decide one way or the other, I don't see any issues with this so I'm approving.

File: client/lib/checkout/payment-methods.tsx
Comment (kanishkdudeja) [reply client/lib/checkout/payment-methods.tsx]: Apologies for the delay, I missed this comment!

I personally lean towards `Unified Payments Interface (UPI)` since I'm presuming users think of the subscription being linked to their UPI handle and don't really remember which payment gateway they initially used to set up the subscription.

File: client/my-sites/email/titan-set-up-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/email/titan-set-up-thank-you/index.tsx]: @billrobbins Looks like the branch needs to be rebased. These changes in `email/titan-set-up-thank-you/index.tsx` have already been merged: https://github.com/Automattic/wp-calypso/pull/88390/files#diff-4de648f48d7bc4d2c6c589e290223d57cea6d434906113c36e52cc88b2877c0f

Comment (wongasy) [reply client/my-sites/email/titan-set-up-thank-you/index.tsx]: @billrobbins Following up to make sure this ping did not fall through the cracks :)

Comment (billrobbins) [reply client/my-sites/email/titan-set-up-thank-you/index.tsx]: Sorry @wongasy.  I'm getting this fixed up today.  Thanks for the ping 👍

**Pull request merged**
